Added troubleshooting and tips tricks content (#15212)
ansible/awx: docs/docsite/rst/administration/containers_instance_groups.rst

   single: container groups
   pair: containers; instance groups

> AWX supports :term:`Container Groups`, which allow you to execute jobs in AWX regardless of whether AWX is installed as a standalone, in  a virtual environment, or in a container. Container groups act as a pool of resources within a virtual environment. You can create instance groups to point to an OpenShift container, which are job environments that are provisioned on-demand as a pod that exists only for the duration of the playbook run. This is known as the ephemeral execution model and ensures a clean environment for every job run.

Reviewer: "whether AWX is installed as a standalone, in a virtual environment, or in a container."
I'm not sure I understand what this means. For upstream we only support one way of installing AWX, through the operator on k8s clusters. I think we can get rid of this clause.

Reviewer: ```suggestion
AWX supports :term:`Container Groups`, which allow you to execute jobs in pods on Kubernetes or OpenShift clusters. Container groups act as a pool of resources within a virtual environment. These pods are created on-demand and only exist for the duration of the playbook run. This is known as the ephemeral execution model and ensures a clean environment for every job run.
```

.. _ag_customize_pod_spec:

Customize the pod spec
^^^^^^^^^^^^^^^^^^^^^^^^

> AWX provides a simple default pod specification, however, you can provide a custom YAML (or JSON) document that overrides the default pod spec. This field uses any custom fields (i.e. ``ImagePullSecrets``) that can be "serialized" as valid pod JSON or YAML. A full list of options can be found in the `OpenShift documentation <https://docs.openshift.com/online/pro/architecture/core_concepts/pods_and_services.html>`_.

Reviewer 2: ```suggestion
AWX provides a simple default pod specification, however, you can provide a custom YAML (or JSON) document that overrides the default pod spec. This field uses any custom fields (for example, ``ImagePullSecrets``) that can be "serialized" as valid pod JSON or YAML. A full list of options can be found in the `OpenShift documentation <https://docs.openshift.com/online/pro/architecture/core_concepts/pods_and_services.html>`_.
```

	The image used at job launch time is determined by which |ee| is associated with the job. If a Container Registry credential is associated with the |ee|, then AWX will attempt to make a ``ImagePullSecret`` to pull the image. If you prefer not to give the service account permission to manage secrets, you must pre-create the ``ImagePullSecret`` and specify it on the pod spec, and omit any credential from the |ee| used.

.. tip::

> 	In order to override DNS/host entries, add the following entries in the in the ``/etc/hosts`` file of the pod spec via the ``hostAliases`` attribute:

Reviewer: ```suggestion
	In order to override DNS/host entries, use the ``hostAliases`` attribute on the pod spec. When the pod is created, these entries will be added to ``/etc/hosts`` in the container running the job.
```

.. note::

> 	Using a custom pod spec may cause issues on upgrades if the default ``pod_spec`` changes. Most any manifest can be applied to any namespace, with the namespace specified separately, most likely you will only need to override the namespace. Similarly, pinning a default image for different releases of the platform to different versions of the default job runner container is tricky. If the default image is specified in the pod spec, then upgrades do not pick up the new default changes are made to the default pod spec.

Reviewer 2: ```suggestion
	Using a custom pod spec may cause issues on upgrades if the default ``pod_spec`` changes. Most any manifest can be applied to any namespace, with the namespace specified separately, most likely you will only need to override the namespace. Similarly, pinning a default image for different releases of the platform to different versions of the default job runner container is tricky. If the default image is specified in the pod spec, then upgrades do not pick up the new default changes are made to the default pod spec.
```
```suggestion
	Using a custom pod spec may cause issues on upgrades if the default ``pod_spec`` changes. Most any manifest can be applied to any namespace, with the namespace specified separately, most likely you will only need to override the namespace. Similarly, pinning a default image for different releases of the platform to different versions of the default job runner container is tricky. If the default image is specified in the pod spec, then upgrades do not pick up the new default changes that are made to the default pod spec.
```

> 	Using a custom pod spec may cause issues on upgrades if the default ``pod_spec`` changes. Most any manifest can be applied to any namespace, with the namespace specified separately, most likely you will only need to override the namespace. Similarly, pinning a default image for different releases of the platform to different versions of the default job runner container is tricky. If the default image is specified in the pod spec, then upgrades do not pick up the new default changes that are made to the default pod spec.

Reviewer: > Most any manifest

is this supposed to say "Since any manifest"?

.. note::

> 	Despite the fact that customers have custom pod specs, upgrades may be difficult if the default ``pod_spec`` changes. Most any manifest can be applied to any namespace, with the namespace specified separately, most likely you will only need to override the namespace. Similarly, pinning a default image for different releases of the platform to different versions of the default job runner container is tricky. If the default image is specified in the pod spec, then upgrades do not pick up the new default changes are made to the default pod spec.

Reviewer: "Most any manifest can be applied to any namespace, with the namespace specified separately, most likely you will only need to override the namespace."

Should this say "Since any"?

Reviewer: "Despite the fact that customers have custom pod specs, upgrades may be difficult if the default ``pod_spec`` changes"

maybe,

"Note that using custom pod specs may cause issues on upgrade if the default pod spec changes"
